sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala

```diff
@@ -90,6 +88,10 @@ object HiveResult {
           toHiveStructString((key, kType)) + ":" + toHiveStructString((value, vType))
       }.toSeq.sorted.mkString("{", ",", "}")
     case (null, _) => "null"
+    case (d: Date, DateType) =>
+      dateFormatter.format(DateTimeUtils.fromJavaDate(d))
+    case (t: Timestamp, TimestampType) =>
+      DateTimeUtils.timestampToString(timestampFormatter, DateTimeUtils.fromJavaTimestamp(t))
```

Comment:
it's the same as `timestampFormatter.format(DateTimeUtils.fromJavaTimestamp(t))`